front/app/api/files/useAddFile.ts

```diff
@@ -0,0 +1,28 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { CLErrors } from 'typings';
+
+import fetcher from 'utils/cl-react-query/fetcher';
+
+import filesKeys from './keys';
+import { IAddFileProperties, IFile } from './types';
+
+const addFile = async (requestBody: IAddFileProperties) => {
+  return fetcher<IFile>({
+    path: `/files`,
+    action: 'post',
+    body: requestBody,
+  });
+};
+
+const useAddFile = () => {
+  const queryClient = useQueryClient();
+  return useMutation<IFile, CLErrors, IAddFileProperties>({
+    mutationFn: addFile,
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: filesKeys.lists() });
+    },
+  });
+};
+
+// ts-prune-ignore-next
+export default useAddFile;
```

Comment:
Had to use `ts-prune-ignore-next` to fix deadcode issues, since I wanted to add these new endpoints in one PR, and then start using them in subsequent PRs.